src/foundation.cs

```diff
@@ -2130,8 +2130,11 @@ interface NSFormatter : NSCoding, NSCopying {
 		[Export ("isPartialStringValid:newEditingString:errorDescription:")]
 		bool IsPartialStringValid (string partialString, out string newString, out NSString error);
 
+		//[Export ("isPartialStringValid:proposedSelectedRange:originalString:originalSelectedRange:errorDescription:")]
+		//unsafe bool IsPartialStringValid (out string partialString, out NSRange proposedSelRange, string origString, NSRange origSelRange, out NSString error);
+
 		[Export ("isPartialStringValid:proposedSelectedRange:originalString:originalSelectedRange:errorDescription:")]
-		unsafe bool IsPartialStringValid (out string partialString, out NSRange proposedSelRange, string origString, NSRange origSelRange, out NSString error);
+		unsafe bool IsPartialStringValid (ref NSString partialString, out NSRange proposedSelRange, string origString, NSRange origSelRange, out NSString error);
```

Comment:
was `out string` -> `ref NSString` done to please the compiler ? because it seems it will require a cast (and allocation) for most use cases :(

If that's the case then what about changing `out NSString error` to `out string error` ? that will be (most cases) `null` and have a smaller impact